src/streams.rs

```diff
@@ -110,15 +110,19 @@ where
                 let now = SystemTime::now();
                 let update = this
                     .last
-                    .and_then(|last| now.duration_since(last).ok().map(|d| d >= UPDATE_INTERVAL))
+                    .map(|last| now.duration_since(last).unwrap_or_default() >= UPDATE_INTERVAL)
                     .unwrap_or(true);
 
+                *this.transferred += u64::try_from(bytes.len()).unwrap();
+
                 if update && let Some(events) = &this.events {
+                    *this.last = Some(now);
+                    let transferred = std::mem::replace(this.transferred, 0);
```

Comment:
This could use some comments describing what the desired behavior is, or maybe a name change of `TransferStream::transferred`. Is `transferred` meant to be the number of bytes transferred since the last update interval, or since the stream was created? The code looks like it implements the former, but having that name and being a field on the stream (rather than just in `BlockProgress`) makes it unclear.

Reply:
How about a name of `transferred_since_last` and I'll add some comments around it.

Unfortunately the `pin_project_lite::pin_project!` macro chokes on doc strings, but I can at least add some regular comments to the fields.